vault/core.go

```diff
@@ -324,9 +324,9 @@ type Core struct {
 	// These variables holds the config and shares we have until we reach
 	// enough to verify the appropriate root key. Note that the same lock is
 	// used; this isn't time-critical so this shouldn't be a problem.
-	barrierRekeyConfig  *SealConfig
-	recoveryRekeyConfig *SealConfig
-	rekeyLock           sync.RWMutex
+	barrierRotationConfig  *SealConfig
```

Comment:
```suggestion
	rootRotationConfig  *SealConfig
```

This isn't the configuration for the actual barrier keyring rotation (see #1021's terminology), but instead for the root key. While the barrier keyring needs to be involved (the root key is written encrypted via the barrier for standby upgrades, and the keyring needs to be re-encrypted with the new root key), the barrier isn't what is being rotated.

I know this will be a tough change, but I think it'd help with clarity.